UQL/Interpreter.php

```diff
@@ -280,10 +282,15 @@ private function parseValue($value)
      */
     private function callFunction(ASTFunctionCall $functionCall)
     {
+        $functionName = $functionCall->getFunctionName();
+        $arguments = $functionCall->getArguments();
+
+        $arguments[] = new Context($this->dataSource->getEntityClass());
```

Comment:
Yes, I don't like it either, but: 1) we're already creating new objects inside this class (new Parser, new Filter), and 2) this class already has 3 dependencies (actually 4 constructor parameters) and adding another one would break BC.

Sure, 1) having new objects created here without factory doesn't mean we should keep on creating new ones, and 2) if this class has already too many dependencies it means it should be split up into 2 or more classes, but I thought - maybe we should refactor this class in different PR later, now maybe it's better to add this context thingy and implement new functionality in app.

What do you think?